jsparagus/emit/rust.py

```diff
@@ -88,6 +90,22 @@ def indent(writer):
     yield None
     writer.indent -= 1
 
+def extract_ranges(iterator):
+    """Given a sorted iterator of integer, yield the contiguous ranges"""
+    # Identify contiguous ranges of states.
+    ranges = collections.defaultdict(list)
+    for i, v in enumerate(iterator):
+        ranges[v - i].append(v)
```

Comment:
Can you add some comment that explains "if there's contiguous ranges, they're appended to same list" ?
especially, `v - i` needs some explanation.